$OpenBSD: patch-src_libevent-modified_evutil_rand_c,v 1.1 2013/10/23 12:42:20 sthen Exp $

OpenBSD has not needed arc4random_addrandom since 2003, and no longer
provides it.

--- src/libevent-modified/evutil_rand.c.orig	Tue Aug 13 03:32:00 2013
+++ src/libevent-modified/evutil_rand.c	Wed Oct 23 13:10:29 2013
@@ -97,6 +97,11 @@ ev_arc4random_buf(void *buf, size_t n)
 #endif
 }
 
+void
+evutil_secure_rng_add_bytes(const char *buf, size_t n)
+{
+}
+
 #else /* !_EVENT_HAVE_ARC4RANDOM { */
 
 #ifdef _EVENT_ssize_t
@@ -144,18 +149,18 @@ ev_arc4random_buf(void *buf, size_t n)
 	arc4random_buf(buf, n);
 }
 
-#endif /* } !_EVENT_HAVE_ARC4RANDOM */
 
 void
-evutil_secure_rng_get_bytes(void *buf, size_t n)
+evutil_secure_rng_add_bytes(const char *buf, size_t n)
 {
-	ev_arc4random_buf(buf, n);
+	arc4_addrandom((unsigned char*)buf,
+	    n>(size_t)INT_MAX ? INT_MAX : (int)n);
 }
 
+#endif /* } !_EVENT_HAVE_ARC4RANDOM */
+
 void
-evutil_secure_rng_add_bytes(const char *buf, size_t n)
+evutil_secure_rng_get_bytes(void *buf, size_t n)
 {
-	arc4random_addrandom((unsigned char*)buf,
-	    n>(size_t)INT_MAX ? INT_MAX : (int)n);
+	ev_arc4random_buf(buf, n);
 }
-
